src/System.Runtime/tests/System/GC.cs

```diff
@@ -226,4 +226,84 @@ private class TestObject
             }
         }
     }
-}
\ No newline at end of file
+
+    [Fact]
+    public static void ReRegisterForFinalize()
+    {
+        ReRegisterForFinalizeTest.Run();
+    }
+
+    private class ReRegisterForFinalizeTest
+    {
+        public static void Run()
+        {
+            TestObject.Finalized = false;
+            CreateObject();
+
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
```

Comment:
I believe this should be reliable since you are creating the object in CreateObject and upon return of this method I don't see why JIT would ever report it as live (even in dbg build).